CorePayments/src/main/java/com/paypal/android/corepayments/features/eligibility/EligibilityClient.kt

```diff
@@ -0,0 +1,53 @@
+package com.paypal.android.corepayments.features.eligibility
+
+import android.content.Context
+import com.paypal.android.corepayments.CoreConfig
+import com.paypal.android.corepayments.PayPalSDKError
+import com.paypal.android.corepayments.apis.eligibility.EligibilityAPI
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+/**
+ *  Use EligibilityClient to check eligibility for different payment methods.
+ */
+class EligibilityClient internal constructor(
+    private val applicationContext: Context,
+    private val eligibilityAPI: EligibilityAPI,
+    private val dispatcher: CoroutineDispatcher
+) {
+
+    /**
+     * Build an EligibilityClient instance using configuration.
+     *
+     * @param context A reference to an Android context.
+     * @param coreConfig The target configuration (see [CoreConfig]).
+     */
+    constructor(context: Context, coreConfig: CoreConfig) : this(
+        context.applicationContext,
+        EligibilityAPI(coreConfig),
+        Dispatchers.Main
+    )
+
+    /**
+     * Check eligibility for a set of payment methods.
+     *
+     * @param eligibilityRequest A request object (see [EligibilityRequest]).
+     * @param callback A callback to receive a success or error result (see [CheckEligibilityResultListener]).
+     */
+    fun check(
+        eligibilityRequest: EligibilityRequest,
+        callback: CheckEligibilityResultListener
+    ) {
+        CoroutineScope(dispatcher).launch {
+            try {
+                val response =
+                    eligibilityAPI.checkEligibility(applicationContext, eligibilityRequest)
+                callback.onCheckEligibilitySuccess(EligibilityResult(response.isVenmoEligible))
```

Comment:
Do we want to return all of the results here instead of just venmo?

Reply:
Yeah I noticed iOS does this. I'm down to add it in. We should check to see if requesting eligibility for multiple payment methods affects latency though i.e. if it's quicker to ask for Venmo eligibility by itself rather than fetching eligibility for the other payment methods.

Reply:
Returning all other payment method eligibilities here: [f0b95d6](https://github.com/paypal/paypal-android/pull/268/commits/f0b95d69ba3f498f7303643466cccf13d3079b42)